x-pack/plugins/upgrade_assistant/server/lib/es_deprecations_status/migrations.ts

```diff
@@ -0,0 +1,144 @@
+    resolve_during_rolling_upgrade: resolveDuringUpgrade,
+  } = migrationDeprecation;
+
+  return {
+    index: indexName,
+    type: deprecationType,
+    details,
+    message,
+    url,
+    isCritical: level === 'critical',
+    correctiveAction: getCorrectiveAction(message, metadata, indexName),
+    resolveDuringUpgrade,
+  };
+};
+export const getEnrichedDeprecations = async (
+  dataClient: IScopedClusterClient
+): Promise<EnrichedDeprecationInfo[]> => {
+  const deprecations = await dataClient.asCurrentUser.migration.deprecations();
+  const systemIndices = await getESSystemIndicesMigrationStatus(dataClient.asCurrentUser);
+
+  const systemIndicesList = convertFeaturesToIndicesArray(systemIndices.features);
+
+  const indexSettingsIndexNames = Object.keys(deprecations.index_settings).map(
+    (indexName) => indexName!
+  );
+  const indexSettingsIndexStates = indexSettingsIndexNames.length
+    ? await esIndicesStateCheck(dataClient.asCurrentUser, indexSettingsIndexNames)
+    : {};
+
+  return Object.entries(deprecations)
+    .map(([deprecationType, deprecationsByType]) => {
+      switch (deprecationType as keyof EsDeprecations) {
+        case 'index_settings': {
+          return Object.keys(deprecations.index_settings)
+            .filter((indexName) => {
+              // filter out system indices
+              return !systemIndicesList.includes(indexName!);
+            })
+            .map((indexName) => {
+              const enrichedDeprecation = createBaseMigrationDeprecation(deprecationsByType, {
+                indexName,
+                deprecationType,
+              });
+
+              // If we have found deprecation information for index/indices
+              // check whether the index is open or closed.
+              if (enrichedDeprecation.correctiveAction?.type === 'reindex') {
+                enrichedDeprecation.correctiveAction.blockerForReindexing =
+                  indexSettingsIndexStates[enrichedDeprecation.index!] === 'closed'
+                    ? 'index-closed'
+                    : undefined;
+              }
+
+              return enrichedDeprecation;
+            })
+            .flat();
+        }
+        case 'data_streams': {
+          return Object.entries(deprecationsByType).map(([indexName, dataSteamDeprecations]) => {
+            return dataSteamDeprecations.map((depractionData) =>
+              createBaseMigrationDeprecation(depractionData, { indexName, deprecationType })
+            );
+          });
+        }
+        case 'ml_settings':
+        case 'node_settings':
+        case 'cluster_settings': {
+          return deprecationsByType.map((depractionData) =>
+            createBaseMigrationDeprecation(depractionData, { deprecationType })
+          );
+        }
+        default: {
+          throw new Error(`Unknown ES deprecation type "${deprecationType}"`);
```

Comment:
I think throwing here makes sense. The alternative is to ignore/log/surface this somehow, so this seems simplest. Just a bit scary given that this object might change at an unknown point again 😄